src/components/views/dialogs/spotlight/SpotlightDialog.tsx

```diff
@@ -538,10 +550,15 @@ const SpotlightDialog: React.FC<IProps> = ({ initialText = "", initialFilter = n
             metricsViaKeyboard: viaKeyboard,
             room_id: room.roomId,
             room_alias: room.roomAlias,
-            auto_join: room.autoJoin,
+            auto_join: room.autoJoin && !canAskToJoin(room.joinRule),
```

Comment:
What if I am allowed to join the room by the [`knock_restricted` rule](https://spec.matrix.org/v1.8/client-server-api/#mroomjoin_rules)?

Reply:
Is `knock_restricted` already implemented in element-web? I consider it to be handled separately.

Reply:
Okay